src/mame/tatsumi/tatsumi_rotating_sprites.cpp

```diff
@@ -0,0 +1,552 @@
+// license:BSD-3-Clause
+// copyright-holders:Bryan McPhail, Angelo Salese
+
+#include "emu.h"
+#include "tatsumi_rotating_sprites.h"
+#include "screen.h"
+
+DEFINE_DEVICE_TYPE(TATSUMI_ROTATING_SPRITES, tatsumi_rotating_sprites_device, "tatsumi_rotating_spr", "Tatsumi Rotating Sprites")
+DEFINE_DEVICE_TYPE(TATSUMI_ROTATING_SPRITES_BIGPAL, tatsumi_rotating_sprites_bigpal_device, "tatsumi_rotating_spr_bigpal", "Tatsumi Rotating Sprites (larger palette)")
+
+
+tatsumi_rotating_sprites_device::tatsumi_rotating_sprites_device(const machine_config &mconfig, device_type type, const char *tag, device_t *owner, uint32_t clock)
+	: device_t(mconfig, type, tag, owner, clock)
+	, device_gfx_interface(mconfig, *this)
+	, m_fakepalette(*this, "fakepalette")
+	, m_basepalette(*this, finder_base::DUMMY_TAG)
+	, m_spriteram(*this, finder_base::DUMMY_TAG)
+	, m_sprites_l_rom(*this, "sprites_l")
+	, m_sprites_h_rom(*this, "sprites_h")
+{
+}
+
+tatsumi_rotating_sprites_device::tatsumi_rotating_sprites_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock)
+	: tatsumi_rotating_sprites_device(mconfig, TATSUMI_ROTATING_SPRITES, tag, owner, clock)
+{
+	m_rom_clut_size = 0x800;
+}
+
+tatsumi_rotating_sprites_bigpal_device::tatsumi_rotating_sprites_bigpal_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock)
+	: tatsumi_rotating_sprites_device(mconfig, TATSUMI_ROTATING_SPRITES_BIGPAL, tag, owner, clock)
+{
+	m_rom_clut_size = 0x1000;
+}
+
+void tatsumi_rotating_sprites_device::common_init()
+{
+	m_rom_clut_offset = memregion("sprites_l")->bytes() - m_rom_clut_size;
+
+	m_shadow_pen_array = make_unique_clear<uint8_t[]>(8192);
+	m_temp_bitmap.allocate(512, 512);
+}
+
+static const gfx_layout spritelayout =
+{
+	8,8,
+	RGN_FRAC(1,1),
+	4,
+	{ STEP4(0,1) },
+	{ 8,12,0,4, 24,28,16,20 },
+	{ STEP8(0,4*8) },
+	32*8
+};
+
+GFXDECODE_MEMBER( tatsumi_rotating_sprites_device::gfxinfo )
+	GFXDECODE_DEVICE("sprites_l", 0, spritelayout, 0, 256)
+	GFXDECODE_DEVICE("sprites_h", 0, spritelayout, 0, 256)
+GFXDECODE_END
+
+GFXDECODE_MEMBER( tatsumi_rotating_sprites_bigpal_device::gfxinfo_big )
+	GFXDECODE_DEVICE("sprites_l", 0, spritelayout, 0, 512)
+	GFXDECODE_DEVICE("sprites_h", 0, spritelayout, 0, 512)
+GFXDECODE_END
+
+void tatsumi_rotating_sprites_device::device_start()
+{
+	common_init();
+	decode_gfx(gfxinfo);
+}
+
+void tatsumi_rotating_sprites_bigpal_device::device_start()
+{
+	common_init();
+	decode_gfx(gfxinfo_big);
+}
+
+void tatsumi_rotating_sprites_device::device_reset()
+{
+}
+
+void tatsumi_rotating_sprites_device::device_add_mconfig(machine_config &config)
+{
+	PALETTE(config, m_fakepalette).set_format(palette_device::xRGB_555, 4096); // 4096 arranged as series of CLUTs
+}
+
+void tatsumi_rotating_sprites_bigpal_device::device_add_mconfig(machine_config &config)
+{
+	PALETTE(config, m_fakepalette).set_format(palette_device::xRGB_555, 8192); // 8192 arranged as series of CLUTs
+}
```

Comment:
You don’t need to override this in the derived class, you can just use `m_rom_clut_size * 2` for the number of entries.